superset/datasets/dao.py

```diff
@@ -190,28 +185,37 @@ def update_columns(
         - If there are extra columns on the metadata db that are not defined on the List
         then we delete.
         """
+
+        column_by_id = {column.id: column for column in model.columns}
         new_columns = []
+
         for column in property_columns:
             column_id = column.get("id")
 
             if column_id:
-                column_obj = db.session.query(TableColumn).get(column_id)
-                column_obj = DatasetDAO.update_column(column_obj, column, commit=commit)
+                column_obj = DatasetDAO.update_column(
+                    column_by_id.get(column_id),
+                    commit=False,
```

Comment:
No need to commit when updating the column. It can all be committed when the dataset is updated.